File: WebClientPy/WebClient.py
import os, http.client, ssl, sys

hostname = ("localhost", 443) # default
if len(sys.argv) > 1:
    host_port = sys.argv[1].split(":")
    hostname = (host_port[0], int(host_port[1]))

# file paths relative to this script
# TODO: Figure out how to load client cert from Windows cert store (CNG) instead of file
CERT_FILE = os.path.join(os.path.dirname(__file__), '..\\TestCertificates\\ClientCert.pem')

# configure client certificate
context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
context.load_default_certs()
context.load_cert_chain(certfile=CERT_FILE)

# submit HTTP request
conn = http.client.HTTPSConnection(hostname[0], port=hostname[1], context=context)
conn.request('GET', '/')

# print response
r = conn.getresponse()
print(r.read())

File: WebClientGo/WebClient.go
package main

import (
    "crypto/tls"
    "fmt"
    "io/ioutil"
    "log"
    "net/http"
)

func main() {
    // TODO: Figure out how to load client cert from Windows cert store (CNG) instead of file
    certificate, err := tls.LoadX509KeyPair("../TestCertificates/ClientCert.pem", "../TestCertificates/ClientCert.pem")
    if err != nil {
        log.Fatalf("could not load certificate: %v", err)
    }

    client := http.Client{
        Transport: &http.Transport{
            TLSClientConfig: &tls.Config{
                Certificates: []tls.Certificate{certificate},
            },
        },
    }

    resp,err := client.Get("https://localhost")
    if err != nil {
        log.Fatalf("error making get request: %v", err)
    }

    body,err := ioutil.ReadAll(resp.Body)
    if err != nil{
        log.Fatalf("error reading response: %v", err)
    }
    fmt.Println(string(body))
}
